### ISSUE TITLE
Move to pygeos for minimum bounding rectangle (rotated) and minimum bounding circle

### ISSUE THREAD

**Issue by ljwolf** [pysal/pointpats#75]
This would address #70, #69

**Comment by martinfleis** [comment]
I would wait with minimum bounding rectangle for GEOS 3.12 that will contain a fixed algorithm for this, resulting in identical geoms OpenCV produces. The circle can be done now, we just need to extract the tuple of x, y and radius from the actual polygon (as centroid and a distance to first point).

**Comment by jGaboardi** [comment]
With #162 in place, [`shapely>2.1`](https://github.com/pysal/pointpats/blob/a1a36422712bb16bee550c023462ebf03dfadfd1/pyproject.toml#L36) is now the oldest version supported. So would this be a "simple" refactor of [`minimum_bounding_circle()`](https://github.com/pysal/pointpats/blob/a1a36422712bb16bee550c023462ebf03dfadfd1/pointpats/centrography.py#L910C1-L1000C41) stuff?

**Comment by martinfleis** [comment]
For rotated minimum bounding circle, the Skyum used in pointpats does not match the GEOS outcome. Ours is actually smaller.

Taking the point pattern from tests:

```py
import geopandas as gpd
import numpy as np
import shapely

points = np.array([(1, 2), (2, 3), (2, 2), (2, 1), (3, 4), (3, 3), (3, 1), (4, 4)])
geoms = gpd.GeoSeries.from_xy(*points.T)

# skyum outcome
x = 2.642857142857143
y = 2.7857142857142856
r = 1.821078397711709

ax = geoms.plot(figsize=(12, 12))
gpd.GeoSeries([shapely.Point(x, y).buffer(r)]).boundary.plot(ax=ax, color='r', linewidth=1)
geoms.to_frame().dissolve().minimum_bounding_circle().boundary.plot(ax=ax, color='g', linewidth=1)
```

<img width="1011" height="984" alt="Image" src="https://github.com/user-attachments/assets/9822f071-6a65-40a5-8473-b15cf411ea81" />

```py
>>> gpd.GeoSeries([shapely.Point(x, y).buffer(r)]).area
0    10.401819
dtype: float64

>>> geoms.to_frame().dissolve().minimum_bounding_circle().area
0    10.550485
dtype: float64
```

**Comment by ljwolf** [comment]
Interesting...

Yes we need to move. This may be a winding issue, akin to the issues we've had before with atan2 vs arctan.

**Comment by martinfleis** [comment]
I was thinking reporting this to GEOS, given it seems that their algorithm is not the minimum in all cases.

**Comment by martinfleis** [comment]
GEOS, according to the source code of JTS, follows Rokne (1991) - https://doi.org/10.1016/B978-0-08-050754-5.50014-1. Our comes from Skyum (1991) - https://doi.org/10.1016/0020-0190(91)90030-L . These are quite different implementations from about the same time. Neither refers to the other.

Skyum is O(n log n) while Rokne reports O(n**2) so Skyum seems to be not only smaller but also more efficient.

**Comment by ljwolf** [comment]
Yes, I picked Skyum because it was a "simple and fast" algorithm. Many hulling/compactness measures only need to consider points on convex hull.

Anyway the issue I think is in the comment `# confusing as hell defaults`. I have it recovering the same MBC as `shapely` by swapping the lexsort order there and in the `numba` implementation.

**Comment by ljwolf** [comment]
> I was thinking reporting this to GEOS, given it seems that their algorithm is not the minimum in all cases.

Would you have an example we can test for the Skyum implementation?

**Comment by martinfleis** [comment]
> Would you have an example we can test for the Skyum implementation?

The case above, no? That is when Skyum is smaller. The x, y, r are coming from pointpats code.

> I have it recovering the same MBC as shapely by swapping the lexsort order there and in the numba implementation.

But is it necessarily a better one?

**Comment by ljwolf** [comment]
> is it necessarily a better one

In the example above, our minimum circle is not bounding. The point at (2,1) is not covered. So, yes, I think so. If there is a situation where shapely still recovers a different circle beyond this, I would like to know!

**Comment by martinfleis** [comment]
> The point at (2,1) is not covered. 

of course... sorry, I missed that.

> If there is a situation where shapely still recovers a different circle beyond this, I would like to know!

Not aware of one.

**Comment by jGaboardi** [comment]
So am I understanding correctly here that the skyum implementation is *much* more efficient, but does not guarantee a MBC (as the plot above shows)?

**Comment by ljwolf** [comment]
In theory, Skyum's algorithm is a more performant (worst case) MBC algorithm than Shapley has. But, 
The implementation was incorrect. Sometimes, it would fail and return too small a circle. #206 fixes this.
